egs/lre07/v1/lid/train_ivector_extractor_dnn.sh

```diff
@@ -0,0 +1,180 @@
+  echo "main options (for others, see top of script file)"
+  echo "  --config <config-file>                           # config containing options"
+  echo "  --cmd (utils/run.pl|utils/queue.pl <queue opts>) # how to run jobs."
+  echo "  --num-iters <#iters|10>                          # Number of iterations of E-M"
+  echo "  --nj <n|10>                                      # Number of jobs (also see num-processes and num-threads)"
+  echo "  --num-processes <n|4>                            # Number of processes for each queue job (relates"
+  echo "                                                   # to summing accs in memory)"
+  echo "  --num-threads <n|4>                              # Number of threads for each process (can't be usefully"
+  echo "                                                   # increased much above 4)"
+  echo "  --stage <stage|-4>                               # To control partial reruns"
+  echo "  --num-gselect <n|20>                             # Number of Gaussians to select using"
+  echo "                                                   # diagonal model."
+  echo "  --sum-accs-opt <option|''>                       # Option e.g. '-l hostname=a15' to localize"
+  echo "                                                   # sum-accs process to nfs server."
+  exit 1;
+fi
+
+fgmm_model=$1
+nnet=$2
+data=$3
+data_dnn=$4
+dir=$5
+
+srcdir=$(dirname $fgmm_model)
+
+for f in $fgmm_model $data/feats.scp ; do
+  [ ! -f $f ] && echo "No such file $f" && exit 1;
+done
+
+# Set various variables.
+mkdir -p $dir/log
+nj_full=$[$nj*$num_processes]
+sdata=$data/split$nj_full;
+utils/split_data.sh $data $nj_full || exit 1;
+
+sdata_dnn=$data_dnn/split$nj_full;
+utils/split_data.sh $data_dnn $nj_full || exit 1;
+
+
+splice_opts=`cat exp/nnet//splice_opts 2>/dev/null` # frame-splicing options           
+
+#parallel_opts="-pe smp $[$num_threads*$num_processes]"
+parallel_opts="-pe smp $num_processes"
+## Set up features.
+
+feats="ark,s,cs:apply-cmvn-sliding --norm-vars=false --center=true --cmn-window=300 scp:$sdata/JOB/feats.scp ark:- | add-deltas-sdc ark:- ark:- | select-voiced-frames ark:- scp,s,cs:$sdata/JOB/vad.scp ark:- |"
+
+nnet_feats="ark,s,cs:apply-cmvn-sliding --center=true scp:$sdata_dnn/JOB/feats.scp ark:- |"
+
+
+# Initialize the i-vector extractor using the FGMM input
+if [ $stage -le -2 ]; then
+  cp $fgmm_model $dir/final.ubm || exit 1;
+  $cmd $dir/log/convert.log \
+    fgmm-global-to-gmm $dir/final.ubm $dir/final.dubm || exit 1;
+  $cmd $dir/log/init.log \
+    ivector-extractor-init --ivector-dim=$ivector_dim --use-weights=$use_weights \
+     $dir/final.ubm $dir/0.ie || exit 1;
+fi 
+
+# Do Gaussian selection and posterior extracion
+
+if [ $stage -le -1 ]; then
+  echo $nj_full > $dir/num_jobs
+  echo "$0: doing DNN posterior computation"
+  $cmd JOB=1:$nj_full $dir/log/post.JOB.log \
+  nnet-am-compute --apply-log=true --use-gpu=yes --chunk-size=400 $nnet "$nnet_feats" ark:- \
```

Comment:
Sorry to reverse what I asked you to do earlier, but I think it will be easier if you change this back to --use-gpu=no. To make GPU support in this script consistent with other Kaldi scripts, you would need to make additional changes that go beyond just setting that one option.

However, you might want to leave a comment in there that says that --use-gpu=yes enables GPU support